src/arrayMethodReduce.js

```diff
@@ -1,11 +1,20 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acum = initialValue;
+
+    if (arguments.length === 1 && typeof this[0] === 'number') {
```

Comment:
The condition for checking the type of the first element in the array is not enough. It's not only numbers and strings that can be reduced, any type can be reduced. This can lead to unpredictable behavior.